packages/ragbits-core/src/ragbits/core/embeddings/litellm.py

```diff
@@ -8,17 +8,31 @@
     EmbeddingResponseError,
     EmbeddingStatusError,
 )
+from ragbits.core.options import Options
+from ragbits.core.types import NOT_GIVEN, NotGiven
 
 
-class LiteLLMEmbeddings(Embeddings):
+class LiteLLMEmbeddingsOptions(Options):
+    """
+    Dataclass that represents available call options for the LiteLLMEmbeddingClient client.
+    Each of them is described in the [LiteLLM documentation](https://docs.litellm.ai/docs/embedding/supported_embedding#optional-litellm-fields).
+    """
+
+    dimensions: int | None | NotGiven = NOT_GIVEN
+    timeout: int | None | NotGiven = NOT_GIVEN
+    user: str | None | NotGiven = NOT_GIVEN
+    encoding_format: str | None | NotGiven = NOT_GIVEN
+
+
+class LiteLLMEmbeddings(Embeddings[LiteLLMEmbeddingsOptions]):
     """
     Client for creating text embeddings using LiteLLM API.
     """
 
     def __init__(
         self,
         model: str = "text-embedding-3-small",
-        options: dict | None = None,
+        options: LiteLLMEmbeddingsOptions | None = None,
```

Comment:
How would you use it when instantiating using with `Embeddings.from_config(...)` (note that the argument to `from_config` only contains simple types, serialized from an external yaml/json file)?

The same question to other classes.

See `LLM.from_config` to compare how `from_config` looks like for classes that used class based options before.

Reply:
Right, I'll add `from_config` that use class based option to `Embeddings` 

> The same question to other classes.

Which classes? 

Currently options will be supported in `LLMS`, `VectorStores` (already covered) and `Embeddings` (I need to add custom `from_config`) and `Rerankers` (in progress)

Do yo mean any other classes?

Reply:
What I meant is that it's a general issue that probably needs a general solution - `Embeddings` is an example, but I would try thinking of a solution that covers the current and feature cases. For example add an optional class property `options_cls` directly to `WithConstructionConfig` and change `WithConstructionConfig.from_config` to serialize `options` using `options_cls` if it is specified.